main.cc

```diff
@@ -1427,8 +1427,7 @@ To start the scylla server proper, simply invoke as: scylla server (or just scyl
                   // This means only the local memtables need to be flushed.
                   db.local().flush_all_memtables().get();
                   supervisor::notify("replaying schema commit log - removing old commitlog segments");
-                  //FIXME: discarded future
-                  (void)sch_cl->delete_segments(std::move(paths));
+                  sch_cl->delete_segments(std::move(paths)).get();
```

Comment:
why was this future ignored before?

Reply:
I assume to avoid delaying startup due to deletions. However, my rationale for _not_ doing so is that a.) the time to delete and/or rename X files should be negligible. Esp. if in parallel, as this will more be. 
b.) We literally have tests that do delays in footprint measurements to deal with this background processing, i.e. added flakiness. Adding a few ms for recover boot seems a small price to pay for essentially consistency in behaviour/footprint.

Reply:
Make sense, I was just curious.